src/utils.js

```diff
@@ -0,0 +1,94 @@
+const fs = require("fs");
+const child_process = require("child_process");
+const path = require("path");
+const glob = require("glob");
+const semver = require("semver");
+
+function getTags() {
+    return child_process.execSync(`git tag`, {encoding: "utf8"}).split("\n").filter(Boolean);
+}
+
+function getWorkspaces() {
+    const pkgJson = JSON.parse(fs.readFileSync("package.json", "utf8"));
+    return pkgJson.workspaces.reduce((result, wsGlob) => {
+        return result.concat(glob.sync(wsGlob));
+    }, []);
+}
+
+// We assume that if a there's a tag for the current version of a package that
+// it's been published already.  In the future we should check if that version
+// exists on npm and attempt to republish the package if it isn't.
+// See https://github.com/Khan/antwerp/issues/3.
+function getCommands() {
+    const commands = [];
+    const allTags = getTags();
+    for (const workspace of getWorkspaces()) {
+        const pkgPath = path.join(workspace, "package.json");
+        const pkg = JSON.parse(fs.readFileSync(pkgPath, "utf8"));
+        if (pkg.private) {
+            continue;
+        }
+        const tag = `${pkg.name}@${pkg.version}`;
+        if (!allTags.includes(tag)) {
+            commands.push(`git tag ${tag}`);
+            // We use npm instead of yarn for publishing b/c yarn publish
+            // bumps the version automatically
+            commands.push(`npm publish --access=public ${workspace}`);
+        }
+    }
+
+    return commands;
+}
+
+function runChecks(verbose = false) {
+    const execOptions = {
+        encoding: "utf8",
+    };
+        
+    const pkgVerMap = {};
+    const depMap = {};
+    const pkgJsonMap = {};
+    
+    for (const workspace of getWorkspaces()) {
+        const pkgPath = path.join(workspace, "package.json");
+        if (!fs.existsSync(pkgPath)) {
+            throw new Error(`no package.json for ${workspace}`);
+        }
+        const pkgJson = JSON.parse(fs.readFileSync(pkgPath));
+        pkgVerMap[pkgJson.name] = pkgJson.version;
+        depMap[pkgJson.name] = pkgJson.dependencies || {};
+        pkgJsonMap[pkgJson.name] = pkgJson;
+    }
```

Comment:
So, this loop is ensuring that even though the dependency may not exist yet (because it's on a version of the package that isn't released), it will when `antwerp` is finished. Right?

Can we get this loop in its own method to make that a little clearer?